### ISSUE TITLE
Can't focus or close windows when nativeWindowOpen is true

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#18513]
### Preflight Checklist
* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details
* **Electron Version:** 4.0.4
* **Operating System:** Windows 10
* **Last Known Working Electron version:** n/a

### Expected Behavior
I open a native window with `w = window.open(...)`. `w.focus()` focuses the window. `w.close()` closes the window.

### Actual Behavior
`w.focus()` doesn't focus the window and `w.close()` doesn't close the window if it's cross-domain.

### To Reproduce
Seems related to [this issue](https://github.com/electron/electron/issues/8969), but I can reproduce it without sandboxing. Turning `nativeWindowOpen` off resolves the issue, so it appears related to that.

Download the electron quick start app and:

main.js:
```
const {app, BrowserWindow} = require('electron')

function createWindow () {
  var mainWindow = new BrowserWindow({webPreferences: {nativeWindowOpen: true}});
  mainWindow.loadFile('index.html');
}

app.on('ready', createWindow)
```

index.html's body:
```
<webview id="w" src="about:blank;" allowpopups></webview>
<script>
   require('./renderer.js')
</script>
```

renderer.js:
```
setTimeout(function() {
	var w = document.getElementById("w");
	w.loadURL("https://google.com");
	w.openDevTools();
}, 1000); // after webview has loaded
```

After `npm start`ing the app, execute the following in the devtools:

    w = window.open("https://google.com", "a", "width=500");
    w.focus();

Observe that the window doesn't get focus.

Similarly, `w.close()` doesn't close the window if it points to a cross-domain site, but it sets `closed` to `true`.

Turning off `nativeWindowOpen` makes it work.

Tried in Chrome, and it works there.

Electron 5 and 6 also appear to be broken, while Canary (which uses the Chromium that Electron 6 uses I believe) works.

Same issue when using BrowserView instead of webview.

**Comment by pushkin-** [comment]
@codebytere Have you investigated enough to know if this is an Electron issue that is likely to be fixed, or is this more likely to be closed as some Chromium issue. If it's likely to be closed, I want to know so I can implement a workaround in my app. Thank you

**Comment by pushkin-** [comment]
@codebytere Seeing as you've self-assigned this issue, can you comment on its status? Thank you

**Comment by codebytere** [comment]
@pushkin- this should be resolved by https://github.com/electron/electron/pull/19703

**Comment by loc** [comment]
The PR linked above won't fix it, but #21416 should.

**Comment by pushkin-** [comment]
@loc , FYI I can still repro this with Electron 8.0.1 (which should contain that PR). I guess I'll wait on [this PR](https://github.com/electron/electron/pull/19703) to get merged in. @sofianguy Could we get the Electron 8.0.1 label added?

**Comment by pushkin-** [comment]
@loc @nornagon  It seems like the `close` function not working issue is fixed (though it seems to be fixed in Electron 9 which doesn't have the linked PR).

The focus issue persists. I have opened a new issue for it [here](https://github.com/electron/electron/issues/25578)
